cookbooks/aws-parallelcluster-config/templates/default/init/fetch_and_run.erb

```diff
@@ -21,22 +21,40 @@ function error_exit () {
   script=`basename $0`
   echo "parallelcluster: ${script} - $1"
   logger -t parallelcluster "${script} - $1"
+  case $custom_script in
+    OnNodeStart|OnNodeConfigured)
+      echo "$1 Please check /var/log/cfn-init.log in the head node, or check the cfn-init.log in CloudWatch logs. Please refer to https://docs.aws.amazon.com/parallelcluster/latest/ug/troubleshooting-v3.html#troubleshooting-v3-get-logs for more details on ParallelCluster logs." > /var/log/parallelcluster/bootstrap_error_msg
+      ;;
+  esac
   exit 1
 }
 
+# URLs longer than the limit will be trimmed by replacing the middle part with ellipsis
+function process_url (){
+  url=$1
+  if [ ${#url} -gt $url_len_limit ]; then
+    head_len=$(expr $url_len_limit / 2)
+    tail_len=$(expr $url_len_limit - $head_len)
+    echo "$(echo $url | head --bytes=${head_len})......$(echo $url | tail --bytes=${tail_len})"
+  else
+    echo $url
+  fi
+}
+
 function download_run (){
   url=$1
   shift
   scheme=$(echo "${url}"| cut -d: -f1)
+  url_processed="$(process_url "$url")"
   tmpfile=$(mktemp)
   trap "/bin/rm -f $tmpfile" RETURN
   if [ "${scheme}" == "s3" ]; then
-    <%= node['cluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} s3 cp ${url} - > $tmpfile || return 1
+    <%= node['cluster']['cookbook_virtualenv_path'] %>/bin/aws --region ${cfn_region} s3 cp ${url} - > $tmpfile || error_exit "Failed to download ${custom_script} script ${url_processed} using aws s3, return code: $?."
```

Comment:
[Question] Is inclusion of the URL something we need to address for security?

Reply:
This PR merge the existing comments from feature branch to the develop branch. I will create another PR to address the security.